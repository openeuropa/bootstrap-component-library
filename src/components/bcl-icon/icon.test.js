import { merge, renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import demoData from "@openeuropa/bcl-data-icon/data";

const template = "@oe-bcl/bcl-icon/icon.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);
const sizes = ["2xs", "xs", "s", "m", "l", "xl", "2xl", "fluid"];
const transformations = [
  "rotate-90",
  "rotate-180",
  "rotate-270",
  "flip-horizontal",
  "flip-vertical",
];

describe("OE - Icon", () => {
  sizes.forEach((size) => {
    test(`${size} renders correctly`, () => {
      expect.assertions(1);

      return expect(
        render({ ...demoData, icon_size: size })
      ).resolves.toMatchSnapshot();
    });
  });

  transformations.forEach((transformation) => {
    test(`${transformation} renders correctly`, () => {
      expect.assertions(1);

      return expect(
        render({ ...demoData, icon_transformation: transformation })
      ).resolves.toMatchSnapshot();
    });
  });
});
