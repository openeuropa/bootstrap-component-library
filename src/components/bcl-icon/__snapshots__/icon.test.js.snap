// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Icon 2xl renders correctly 1`] = `
<jest>
  <svg
    class="bi icon--s"
  >
    <use
      xlink:href="/icons.svg#eye-slash"
    />
  </svg>
</jest>
`;

exports[`OE - Icon 2xs renders correctly 1`] = `
<jest>
  <svg
    class="bi icon--s"
  >
    <use
      xlink:href="/icons.svg#eye-slash"
    />
  </svg>
</jest>
`;

exports[`OE - Icon flip-horizontal renders correctly 1`] = `
<jest>
  <svg
    class="bi icon--s"
  >
    <use
      xlink:href="/icons.svg#eye-slash"
    />
  </svg>
</jest>
`;

exports[`OE - Icon flip-vertical renders correctly 1`] = `
<jest>
  <svg
    class="bi icon--s"
  >
    <use
      xlink:href="/icons.svg#eye-slash"
    />
  </svg>
</jest>
`;

exports[`OE - Icon fluid renders correctly 1`] = `
<jest>
  <svg
    class="bi icon--s"
  >
    <use
      xlink:href="/icons.svg#eye-slash"
    />
  </svg>
</jest>
`;

exports[`OE - Icon l renders correctly 1`] = `
<jest>
  <svg
    class="bi icon--s"
  >
    <use
      xlink:href="/icons.svg#eye-slash"
    />
  </svg>
</jest>
`;

exports[`OE - Icon m renders correctly 1`] = `
<jest>
  <svg
    class="bi icon--s"
  >
    <use
      xlink:href="/icons.svg#eye-slash"
    />
  </svg>
</jest>
`;

exports[`OE - Icon rotate-90 renders correctly 1`] = `
<jest>
  <svg
    class="bi icon--s"
  >
    <use
      xlink:href="/icons.svg#eye-slash"
    />
  </svg>
</jest>
`;

exports[`OE - Icon rotate-180 renders correctly 1`] = `
<jest>
  <svg
    class="bi icon--s"
  >
    <use
      xlink:href="/icons.svg#eye-slash"
    />
  </svg>
</jest>
`;

exports[`OE - Icon rotate-270 renders correctly 1`] = `
<jest>
  <svg
    class="bi icon--s"
  >
    <use
      xlink:href="/icons.svg#eye-slash"
    />
  </svg>
</jest>
`;

exports[`OE - Icon s renders correctly 1`] = `
<jest>
  <svg
    class="bi icon--s"
  >
    <use
      xlink:href="/icons.svg#eye-slash"
    />
  </svg>
</jest>
`;

exports[`OE - Icon xl renders correctly 1`] = `
<jest>
  <svg
    class="bi icon--s"
  >
    <use
      xlink:href="/icons.svg#eye-slash"
    />
  </svg>
</jest>
`;

exports[`OE - Icon xs renders correctly 1`] = `
<jest>
  <svg
    class="bi icon--s"
  >
    <use
      xlink:href="/icons.svg#eye-slash"
    />
  </svg>
</jest>
`;
